sentry-android-replay/src/main/java/io/sentry/android/replay/viewhierarchy/ViewHierarchyNode.kt

```diff
@@ -0,0 +1,173 @@
+                    shouldRedact = nodeInfo.isVisibleToUser
+                    nodeInfo.recycle()
+                    if (shouldRedact) {
+                        val bounds = Rect()
+                        val text = view.text.toString()
+                        view.paint.getTextBounds(text, 0, text.length, bounds)
+                        dominantColor = adjustAlpha(view.currentTextColor)
+                        rect = Rect()
+                        view.getGlobalVisibleRect(rect)
+
+                        var textEnd = Int.MIN_VALUE
+                        var textStart = Int.MAX_VALUE
+                        if (view.layout != null) {
+                            for (i in 0 until view.layout.lineCount) {
+                                val min = view.layout.getLineStart(i)
+                                val minPosition = view.layout.getPrimaryHorizontal(min).toInt()
+                                val max = view.layout.getLineVisibleEnd(i)
+                                val maxPosition = view.layout.getPrimaryHorizontal(max).toInt()
+                                if (minPosition < textStart) {
+                                    textStart = minPosition
+                                }
+                                if (maxPosition > textEnd) {
+                                    textEnd = maxPosition
+                                }
+                            }
+                        } else {
+                            textEnd = rect.right - rect.left
+                            textStart = 0
+                        }
+                        // TODO: support known 3rd-party widgets like MaterialButton with an icon
+                        // TODO: also calculate height properly based on text bounds
+                        rect.left += textStart + view.paddingStart
+                        rect.right = rect.left + (textEnd - textStart)
+                    }
+                }
+
+                is ImageView -> {
+                    shouldRedact = isVisible(view) && (view.drawable?.isRedactable() ?: false)
+                    if (shouldRedact) {
+                        dominantColor = adjustAlpha((view.drawable?.pickDominantColor() ?: Color.BLACK))
+                        rect = Rect()
+                        view.getGlobalVisibleRect(rect)
+                    }
+                }
+            }
+            return ViewHierarchyNode(
+                view.x,
+                view.y,
+                view.width,
+                view.height,
+                shouldRedact,
+                dominantColor,
+                rect
+            )
+        }
+
+        private fun Drawable.isRedactable(): Boolean {
+            return when (this) {
+                is InsetDrawable, is ColorDrawable, is VectorDrawable, is GradientDrawable -> false
+                is BitmapDrawable -> !bitmap.isRecycled && bitmap.height > 0 && bitmap.width > 0
```

Comment:
m: I would not use 0 but something a little bigger (im using 10 for iOS), some backgrounds are composed by small tiled images.

Reply:
yeah, good idea 👍